File: ArchitectureRefactoringHelper/Repository/Models/Database/Tool.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Repository.Models.Database;

[Table(Constants.TableNameTool + "s")]
public class Tool
{
    [Key]
    [JsonPropertyName("toolId")]
    public int ToolId { get; set; }

    [Required]
    [JsonPropertyName("identifier")]
    public string Identifier { get; set; }

    [JsonIgnore]
    public int ToolSourceId { get; set; }

    [JsonPropertyName("toolSource")]
    public ToolSource ToolSource { get; set; }

    [JsonPropertyName("toolTypes")]
    public ICollection<ToolType>? ToolTypes { get; set; }
    [JsonPropertyName("approachUsabilities")]
    public ICollection<ApproachUsability>? ApproachUsabilities { get; set; }
}

File: UI/api/repository/models/tool.ts
import { ApproachUsability } from "./approach-usability";
import { ToolSource } from "./tool-source";
import { ToolType } from "./tool-type";

export interface Tool{
    toolId?: number;
    identifier: string;
    toolSource?: ToolSource;
    toolTypes?: null | Array<ToolType>;
    approachUsabilities?: null | Array<ApproachUsability>;
}